src/data/migrations/20190822155841_create_user_steps_completed_table.js

```diff
@@ -0,0 +1,29 @@
+exports.up = function(knex) {
+    return knex.schema.createTable("user_steps_completed", completed => {
+        completed.increments()
+        completed
+            .integer("user_id")
+            .unsigned()
+            .notNullable()
+            .references("id")
+            .inTable("users")
+            .onDelete("CASCADE")
+            .onUpdate("CASCADE")
+        completed
+            .integer("steps_id")
+            .unsigned()
+            .notNullable()
+            .references("id")
+            .inTable("steps")
+            .onDelete("CASCADE")
+            .onUpdate("CASCADE")
+        completed
+            .timestamp("created_at", { useTz: true })
+            .defaultTo(knex.fn.now())
+        completed.unique(["user_id", "steps_id"])
```

Comment:
Same here, each user should only be able to complete each step once? So I made the relationship between user_id and steps_id unique. Again we didn't discuss that as a group, hope it's alright